sdk/storage/ustorageclient/build_all/configs/ustorageclient_functions.cmake

```diff
@@ -0,0 +1,189 @@
+    target_include_directories(${what_is_building}_exe PUBLIC ${sharedutil_include_directories})
+    target_link_libraries(${what_is_building}_exe
+                #add azure_ulib_c back later
+                PRIVATE umock_c ctest testrunnerswitcher azure_iot_ustorageclient
+    )
+
+    #Add test to ctest list
+    add_test(NAME ${what_is_building} COMMAND ${what_is_building}_exe)
+
+endfunction()
+
+#Build Linux unit tests with valgrind if enabled
+function(azstorage_c_linux_unittests_add_exe what_is_building folder)
+
+    #Define target
+    add_executable(${what_is_building}_exe
+        ${${what_is_building}_test_files}
+        ${${what_is_building}_cpp_files}
+        ${${what_is_building}_c_files}
+        ${CMAKE_CURRENT_LIST_DIR}/main.c
+    )
+
+    #Set the output folder for the target
+    set_target_properties(${what_is_building}_exe
+               PROPERTIES
+               FOLDER ${folder})
+
+    #Define executable's compile definitions, linked libraries, and include dirs
+    target_compile_definitions(${what_is_building}_exe PUBLIC USE_CTEST)
+    target_include_directories(${what_is_building}_exe PUBLIC ${sharedutil_include_directories})
+    target_link_libraries(${what_is_building}_exe
+                #add azure_ulib_c back later
+                PRIVATE umock_c ctest testrunnerswitcher azure_iot_ustorageclient
+    )
+
+    #Add test to ctest list
+    add_test(NAME ${what_is_building} COMMAND $<TARGET_FILE:${what_is_building}_exe>)
+
+    #Setup valgrind if applicable and add separate test for valgrind/helgrind
+    if(${run_valgrind})
+        find_program(VALGRIND_FOUND NAMES valgrind)
+        if(${VALGRIND_FOUND} STREQUAL VALGRIND_FOUND-NOTFOUND)
+            message(WARNING "run_valgrind was TRUE, but valgrind was not found - there will be no tests run under valgrind")
+        else()
+            add_test(NAME ${what_is_building}_valgrind COMMAND valgrind                 --gen-suppressions=all --num-callers=100 --error-exitcode=1 --leak-check=full --track-origins=yes $<TARGET_FILE:${what_is_building}_exe>)
+            add_test(NAME ${what_is_building}_helgrind COMMAND valgrind --tool=helgrind --gen-suppressions=all --num-callers=100 --error-exitcode=1 $<TARGET_FILE:${what_is_building}_exe>)
+            add_test(NAME ${what_is_building}_drd      COMMAND valgrind --tool=drd      --gen-suppressions=all --num-callers=100 --error-exitcode=1 $<TARGET_FILE:${what_is_building}_exe>)
+        endif()
+    endif()
+
+endfunction()
+
+function(azstorage_build_c_test_target what_is_building folder)
+    #Include repo directories
+    include_directories(${PROJECT_SOURCE_DIR}/inc)
+    include_directories(${PROJECT_SOURCE_DIR}/config)
+
+    #Set shared utility include directories
+    #add ${AZURE_ULIB_C_INC_FOLDER} back later
+    set(ustorageclient_test_framework_includes ${sharedutil_include_directories} ${MACRO_UTILS_INC_FOLDER} ${UMOCK_C_INC_FOLDER} ${TESTRUNNERSWITCHER_INC_FOLDER} ${CTEST_INC_FOLDER})
```

Comment:
just call include_directories (or better yet target_include_directories) with this list.